src/wallet/wallet.cpp

```diff
@@ -2644,6 +2644,34 @@ bool CWallet::FundTransaction(CMutableTransaction& tx, CAmount& nFeeRet, int& nC
     return true;
 }
 
+OutputType CWallet::TransactionChangeType(const std::vector<CRecipient>& vecSend)
+{
+    // If -changetype is specified, always use that change type.
+    if (g_change_type != OUTPUT_TYPE_NONE) {
```

Comment:
For the future, it might be a good idea add `change_type` and `address_type` fields to CoinControl class, and have this method accept a CoinControl argument, instead of relying on the globals. This would make it easier to ensure that when an RPC is being called that has address type options, that they will take precedence over the globals.

Pieter's suggestion of adding a separate `P2SH_SEGWIT_BUT_NATIVE_WHEN_NATIVE_DESTINATION` enum value instead of assigning multiple meanings to `OUTPUT_TYPE_NONE` is also a good idea https://github.com/bitcoin/bitcoin/pull/12119#issuecomment-357982763.

Other cleanups are also possible. I think a nice refactoring PR that didn't change behavior could be submitted that:

* Changed OutputType from enum to enum class.
* Got rid of `DEFAULT` enum value and just set P2SH_SEGWIT as default during initialization.
* Got rid of `NONE` enum value, replacing it with `PARSE_ERROR` and `P2SH_SEGWIT_BUT_NATIVE_WHEN_NATIVE_DESTINATION` (or something shorter like `MATCH_OTHER_OUTPUTS`).

Reply:
Nice wrap up @ryanofsky. In #12194 I'm adding change type to coin control, however I'm not passing coin control here.